src/async_fuse/memfs/metadata.rs

```diff
@@ -820,87 +845,83 @@ impl DefaultMetaData {
 
     /// Helper function to delete or deferred delete node
     async fn may_deferred_delete_node_helper(&self, ino: INum) -> anyhow::Result<()> {
-        let parent_ino: INum;
-        let node_name: String;
-        let mut deferred_deletion = false;
-        {
+        // TODO: support thread-safe to avoid race condition
+        let (deferred_deletion, parent_ino, node_name) = {
```

Comment:
Can we abstract this block into a function? It's too long

Reply:
Done